volatility3/framework/symbols/linux/utilities/modules.py

```diff
@@ -534,3 +534,98 @@ def validate_alignment_patterns(
             True if all the addresses meet the alignment
         """
         return all(addr % address_alignment == 0 for addr in addresses)
+
+
+class ModuleGathererLsmod(ModuleGathererInterface):
+    """
+    Gathers modules from the main kernel list
+    """
+
+    @classmethod
+    def gather_modules(
+        cls, context: interfaces.context.ContextInterface, kernel_module_name: str
+    ) -> ModuleGathererInterface.gatherer_return_type:
+        yield from Modules.list_modules(context, kernel_module_name)
+
+
+class ModuleGathererSysFs(ModuleGathererInterface):
+    """
+    Gathers modules from the sysfs /sys/modules objects
+    """
+
+    @classmethod
+    def gather_modules(
+        cls, context: interfaces.context.ContextInterface, kernel_module_name: str
+    ) -> ModuleGathererInterface.gatherer_return_type:
+        kernel = context.modules[kernel_module_name]
+
+        sysfs_modules: dict = Modules.get_kset_modules(context, kernel_module_name)
+
+        for m_offset in sysfs_modules.values():
+            yield kernel.object(object_type="module", offset=m_offset, absolute=True)
```

Comment:
But otherwise it would just be one extra call here, to hand back a modinfo rather than a module...